client/components/ellipsis-menu/index.jsx

```diff
@@ -60,18 +52,16 @@ class EllipsisMenu extends Component {
 		}
 	};
 
-	setPopoverContext( popoverContext ) {
-		if ( popoverContext ) {
-			this.setState( { popoverContext } );
-		}
-	}
+	hideMenu = () => this.toggleMenu( false );
+
+	showMenu = () => this.toggleMenu( true );
 
-	toggleMenu( isMenuVisible ) {
+	toggleMenu = isMenuVisible => {
```

Comment:
why is that? is it because it's only called from methods where `this` is right?

Reply:
ah. that was silly of me for missing.

do you see any problem using arrow-functions here? I value the consistency and the safety against accidentally breaking changed in the future.

Reply:
> do you see any problem using arrow-functions here? I value the consistency and the safety against accidentally breaking changed in the future.

I share this controversial view and always use arrow functions

Reply:
> do you see any problem using arrow-functions here?

If there are many instances of the class, each of them has their own instance of the method function, created in constructor, consuming memory and construction time. With a regular class, the method lives just once on the prototype, shared by all instances.

That's the whole difference as far as I know, the performance/memory penalty might or might not matter.

Reply:
> That's the whole difference as far as I know, the performance/memory penalty might or might not matter.

Thats the crux of it.  I believe that until proven that there is a meaningful performance hit, I'd rather write in the "safer" way less likely to create bugs in the future.

Reply:
> I believe that until proven

Yes, I guess we differ in what one considers a "default" and what is an exception that needs to bear the burden of proof.

Is a regular class method the default and the arrow method needs to be justified by actually using it in a way that requires binding?

Or is it an arrow method and the regular one needs to prove that it's meaningfully faster?

Reply:
My take is that the arrow-methods are the "fix" to confusing and easy-to-misuse dynamic scoping of `this` in Javascript, so my default is arrow-functions. we _won't_ be surprised by an unexpected `this`

I'll try to pull the changes out of this PR in any case - it's a tangential change just because I was in the file

Reply:
> Or is it an arrow method and the regular one needs to prove that it's meaningfully faster?

Definitely not about faster to me; it's about not accidentally breaking things with no obvious reason why :smile:

Reply:
I wouldn't worry about it unless it shows up in the profiler. I'm mostly annoyed they brought over  `()=>` but not `()->` too :)

Reply:
Here's an idea for a new ESLint rule: for all regular class methods, the only safe operation is to call them directly:
```js
this.toggleMenu();
```
It's not safe to take any "naked" references that are not method calls:
```js
const handler = this.toggleMenu;
addEventListener( 'click', this.toggleMenu );
<button onClick={ this.toggleMenu } />;
```
The rule could warn about usages like these and recommend converting to arrow functions.

React components don't use inheritance and rarely have any methods that are supposed to be a public API (all methods are "private"). So the rule should work flawlessly, shouldn't it?

I'm wondering why it's not already in `eslint-plugin-react`.

Reply:
Is it really a problem though? It seems funny to be concerned about performance for something like this when we wrap almost everything in `connect()` and other HOCs.

You could convince me this is a performance issue if you have stats though. While profiling the work that led me here I saw too much time spend garbage-collecting and adding or freeing up subscribers from `localize()`. I couldn't measure in time if it was significant though, but it threw off the charts.